logging: allow for injection of log level when running unit tests

ianvernon opened the pull request:
Set logging level to error level for the unit test suites. The unit
tests should be debuggable without logs, and the amount of output
from the unit tests makes it difficult to load Travis CI.

Signed-off by: Ian Vernon <ian@cilium.io>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/8597)
<!-- Reviewable:end -->

jrajahalme reviewed (dismissed):
Multiple Suites are run in parallel, so maybe it would be better to set the log level in TestMain()? Also, I don't see the need to restore the Info level. Now, if one Suite finishes, then the info-level logs from the remaining suites start showing up.

Commit:
logging: allow for injection of log level via ldflags

This allows for running unit tests with debugging if a developer desires to do
so. To set the value via `-ldflags` in golang, we have to use a string value,
otherwise we get errors like the following when running `go test`:

```
cannot set with -X: not a var of type string
```

Provide a string representation of the DefaultLogLevel accordingly, and set the
log level of the DefaultLogger in its initialization so that when unit tests
run, the log level injected via `-ldflags` is respected. The default log level
for unit tests is now at error level, which will reduce the verbosity of the
unit tests considerably. We should not rely upon output of our loggers to
debug unit tests anyway.

Signed-off by: Ian Vernon <ian@cilium.io>

aanm commented on Makefile:
```diff
@@ -43,9 +43,13 @@ TEST_LDFLAGS=-ldflags "-X github.com/cilium/cilium/pkg/kvstore.consulDummyAddres
 	-X github.com/cilium/cilium/pkg/testutils.CiliumRootDir=$(ROOT_DIR) \
-	-X github.com/cilium/cilium/pkg/datapath.DatapathSHA=1234567890abcdef7890"
+	-X github.com/cilium/cilium/pkg/datapath.DatapathSHA=1234567890abcdef7890 \
+	-X github.com/cilium/cilium/pkg/logging.DefaultLogLevelStr=error"
```
It would probably be better if we would use a variable to set this value. This variable would be set on line 32 by checking if `LOG_LEVEL` is set. If it isn't, should default to `error`.

ianvernon replied:
Done.

Commit:
test: add environment variable override for log level for unit tests

Users can override the `LOGLEVEL` environment variable to do so, e.g.:

```
TESTPKGS="pkg/kvstore" LOGLEVEL="debug" make unit-tests
```

Signed-off by: Ian Vernon <ian@cilium.io>
